test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const { getContractAt } = ethers;
+const { expect } = require('chai');
+const { getRandomBytes32 } = require('./utils');
+const { deployContract } = require('../scripts/deploy');
+
+/**
+ * Helper function to extract an address from a bytes32 storage slot
+ * @param {string} slotValue - The bytes32 value from storage
+ * @returns {string} The EIP-55 checksummed address
+ */
+function bytes32ToAddress(slotValue) {
+    // Extract the last 20 bytes (40 hex characters) from the bytes32 value
+    const addressHex = '0x' + slotValue.slice(-40);
+    // Convert to EIP-55 checksummed address for consistency
+    return ethers.utils.getAddress(addressHex);
```

Comment:
```suggestion
    const addressHex = '0x' + slotValue.slice(-40);
    return ethers.utils.getAddress(addressHex);
```
superfluous comments, docstring already covers it